File: android_native/app/src/main/java/com/ezbillify/ezworkspace/ui/screens/ActivityScreen.kt
package com.ezbillify.ezworkspace.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.History
import androidx.compose.material.icons.filled.Podcasts
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ezbillify.ezworkspace.theme.AppColors
import com.ezbillify.ezworkspace.theme.InterFontFamily
import com.ezbillify.ezworkspace.ui.components.LiveList

/** Live activity feed — streams the shared `audit_logs` table in realtime. */
@Composable
fun ActivityScreen() {
    Column(Modifier.fillMaxSize()) {
        Row(
            modifier = Modifier.fillMaxWidth().padding(20.dp, 16.dp, 20.dp, 4.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Top,
        ) {
            Column {
                Text("Activity", fontFamily = InterFontFamily, fontWeight = FontWeight.Black, fontSize = 28.sp, color = AppColors.ink)
                Text("Live workspace events", fontFamily = InterFontFamily, fontSize = 12.sp, color = AppColors.muted)
            }
            Icon(Icons.Filled.Podcasts, contentDescription = null, tint = AppColors.primary)
        }
        Box(Modifier.fillMaxSize()) {
            LiveList(table = "audit_logs", icon = Icons.Filled.History, limit = 60)
        }
    }
}

File: flutter_app/lib/screens/activity_screen.dart
import 'package:flutter/material.dart';
import '../theme/app_theme.dart';
import '../widgets/live_list.dart';

/// Live activity feed — streams the shared `audit_logs` table in realtime.
class ActivityScreen extends StatelessWidget {
  const ActivityScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Padding(
          padding: const EdgeInsets.fromLTRB(20, 16, 20, 4),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text('Activity',
                        style: TextStyle(
                          fontSize: AppTheme.fluid(context, 24, 30),
                          fontWeight: FontWeight.w900,
                          color: AppColors.ink,
                        )),
                    Text('Live workspace events', style: TextStyle(fontSize: 12, color: AppColors.muted)),
                  ],
                ),
              ),
              const Icon(Icons.podcasts, color: AppColors.primary),
            ],
          ),
        ),
        const Expanded(
          child: LiveList(table: 'audit_logs', icon: Icons.history, limit: 60),
        ),
      ],
    );
  }
}
